### PR TITLE
Amcl integration test

### PR DESCRIPTION
**Pull request by mhpanah**
Simple integration test that checks to see if amcl is publishing a pose that is a number. Addressing issue #53.

As we continue the re-factorization more tests will be added.  Also, planning to create a separate test to measure algorithms performance.

### CODE REVIEW

File: src/localization/nav2_amcl/CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,10 @@ ament_target_dependencies(${library_name}
   ${dependencies}
 )
 
+find_package(ament_cmake_pytest REQUIRED)
```
Comment (mjeronimo) [inline src/localization/nav2_amcl/CMakeLists.txt]: I think these lines should be inside of the BUILD_TESTING check.

File: src/localization/nav2_amcl/test/integration/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+include_directories( ${PROJECT_SOURCE_DIR}/src )
```
Comment (mjeronimo) [inline src/localization/nav2_amcl/test/integration/CMakeLists.txt]: In the future I'd like to split out the ROS-level integration test into a separate directory. It isn't technically dependent on this implementation. OK for this PR.

File: src/localization/nav2_amcl/test/integration/test_localization_node.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  ~RclCppFixture() { rclcpp::shutdown(); }
+};
+RclCppFixture g_rclcppfixture;
+
+class TestAmclPose : public ::testing::Test
+{
+public:
+  TestAmclPose()
+  {
+    //Initializing amcl_pose x and y to NaN
```
Comment (mjeronimo) [inline src/localization/nav2_amcl/test/integration/test_localization_node.cpp]: Please run the code through cpplint and uncrustify checks. I see a few formatting issues here.

Comment (mhpanah) [reply src/localization/nav2_amcl/test/integration/test_localization_node.cpp]: PR #114 addresses all the ctest issues.

File: src/localization/nav2_amcl/src/amcl_node.cpp [hunk 4]
```diff
@@ -642,7 +642,7 @@ std::lock_guard<std::recursive_mutex> ml(configuration_mutex_);
   // get map via RPC
   auto req = std::make_shared<nav_msgs::srv::GetMap::Request>();
   std::shared_ptr<nav_msgs::srv::GetMap::Response> resp;
-  auto client = this->create_client<nav_msgs::srv::GetMap>("static_map");
+  auto client = this->create_client<nav_msgs::srv::GetMap>("occ_grid");
```
Comment (bpwilcox) [inline src/localization/nav2_amcl/src/amcl_node.cpp]: I think we should make the service name "occ_grid" a variable, likely from a parameter with a default value.

Comment (mhpanah) [reply src/localization/nav2_amcl/src/amcl_node.cpp]: Yes. I agree.

File: src/localization/nav2_amcl/package.xml [hunk 5]
```diff
@@ -35,6 +35,8 @@
   <build_depend>tf2</build_depend>
   <build_depend>nav2_util</build_depend>
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply src/localization/nav2_amcl/package.xml]: Amcl itself is checking for static map.  Yes, we extend the node test to also check for nav2_map_server.

**Pull request merged**